projects/plugins/jetpack/modules/tiled-gallery/tiled-gallery/templates/partials/carousel-image-args.php

```diff
@@ -14,17 +14,16 @@
 if ( defined( 'JSON_HEX_AMP' ) ) {
 	// see shortcodes/slideshow.php
 	// This is nice to have, but not strictly necessary since we use _wp_specialchars() below
-	// phpcs:ignore PHPCompatibility
-	$fuzzy_image_meta = json_encode( array_map( 'strval', $fuzzy_image_meta ), JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT );
+	$fuzzy_image_meta = wp_json_encode( array_map( 'strval', $fuzzy_image_meta ), JSON_HEX_TAG | JSON_HEX_AMP | JSON_HEX_APOS | JSON_HEX_QUOT );
 } else {
-	$fuzzy_image_meta = json_encode( array_map( 'strval', $fuzzy_image_meta ) );
+	$fuzzy_image_meta = wp_json_encode( array_map( 'strval', $fuzzy_image_meta ) );
 }
 ?>
 data-attachment-id="<?php echo esc_attr( $item->image->ID ); ?>"
 data-orig-file="<?php echo esc_url( wp_get_attachment_url( $item->image->ID ) ); ?>"
 data-orig-size="<?php echo esc_attr( $item->meta_width() ); ?>,<?php echo esc_attr( $item->meta_height() ); ?>"
 data-comments-opened="<?php echo esc_attr( comments_open( $item->image->ID ) ); ?>"
-data-image-meta="<?php echo _wp_specialchars( wp_check_invalid_utf8( $fuzzy_image_meta ), ENT_QUOTES, false, true ); ?>"
+data-image-meta="<?php echo esc_attr( _wp_specialchars( wp_check_invalid_utf8( $fuzzy_image_meta ), ENT_QUOTES, false, true ) ); ?>"
```

Comment:
Same as in the previous file.

Reply:
Let me know if it looks like I did this correctly: a4a2c02ee943bb7d90339a0e0b588cfa884f402a

Reply:
LGTM